cli/crates/server/src/codegen_worker.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{event::Event, types::ServerMessage};
```

Comment:
Should this file be named `codegen_server` instead for consistency with the rest of the modules in the crate?

Reply:
Also: should it do more error reporting? It is mostly silent now.